### PR TITLE
Produce artifact

### PR DESCRIPTION
**Pull request by TriplEight**
Closes paritytech/devops#676

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -56,3 +64,22 @@ build-test:
...
+    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/        # i.e. v1.0, v2.1rc1
+    - if: $CI_COMMIT_REF_NAME == "master"
+      when: manual
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                   # PRs
+      when: manual
+  script:
+    - cargo build --release
+  after_script:
+    - mkdir -p ./artifacts/cargo-contract/
+    - cp ${CARGO_TARGET_DIR}/release/cargo-contract ./artifacts/cargo-contract/
```
Comment (ascjones) [inline .gitlab-ci.yml:85]: Should be `canvas-node` rather than `cargo-contract`?

Comment (TriplEight) [reply .gitlab-ci.yml:85]: lols, ehehe, let me fix that :sweat_smile:

**Review by ascjones — COMMENTED** [review]
The build step probably shouldn't show up in the checks? 

image

**Comment by TriplEight** [comment]
> The build step probably shouldn't show up in the checks?

What I did is I put it on manual in the regular PRs. So everyone can have a binary from any build. So you push `>` there and then can download the artifact from gitlab.

But if you do not find it useful, surely can remove it.

**Comment by ascjones** [comment]
I think it's useful, but it does make the pending checks hang :point_up:

**Comment by TriplEight** [comment]
Oh, that's because ` ci/gitlab/gitlab.parity.io `, aka "the entire pipeline" is required.
image

**Comment by TriplEight** [comment]
Also made them allowed to fail so GH will not wait their status. But up to you, still can remove those manual runs from PR and master pipelines.

**Review by ascjones — APPROVED** [review]
LGTM. Though not an expert on all the gitlab CI config magic.

**Pull request merged**